### PR TITLE
feat(typeahead): improved accessibility #582

### LINKED ISSUE #582: feat(typeahead): add accessibility attributes
Since bootstrap components will have a big user base, It is a good time to think of accessibility on all of the components. In particular, this typeahead widget is not currently accessible. When results are show, there is no indication how many results are there and how to navigate the results. 

With screen reader turned on, please look at this example: http://www.bu.edu/webteam/clrux/accessibility/autocomplete/autocomplete.html. 

Also, current implementation doesn't seem to allow overriding the "result" template. I have no way to adding some aria attributes to enable accessibility.

### PR DESCRIPTION
**Pull request by samshay**
should close #582 

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [X] built and tested the changes locally.
 - [ ] added/updated tests ( no test for typeahead)
 - [X] added/updated API documentation.
 - [ ] added/updated demos ( not necessary )

To add accessibility to typeahead, i copy the same modification of ng-bootstrap :
https://github.com/ng-bootstrap/ng-bootstrap/commit/e1fa7a493cfca4677992c7fc60961f19323b829d#diff-2b05df9eab924379f039993fae7b433e

I test this with NVDA and it can voice correctly now.

Reference issue : #582

### CODE REVIEW

**Comment by samshay** [comment]
What's the status of merging pull requests to this repo? I see there are more waiting.

**Review by valorkin — DISMISSED** [review]
looks solid, thanks a lot for this great contribution!

**Review by daniloff200 — DISMISSED** [review]

**Comment by samshay** [comment]
Sorry, but when this pull request will it be merge in develop ?

**Comment by dmitry-zhemchugov** [comment]
Hi @samshay !  I have tested this PR and I think you should rewrite it like accessibility parts in other components to bring the page of component to a single style with others.

For example: 
https://valor-software.com/ngx-bootstrap/#/dropdowns#accessibility
https://valor-software.com/ngx-bootstrap/#/collapse#accessibility
https://valor-software.com/ngx-bootstrap/#/sortable#accessibility

**Comment by samshay** [comment]
Ok, thanks, it's possible to do this with dropdowns, collapse because the code is not into the component, and the aria attribute can be add in client html code.
But for typeahead, the code (html) is inside the component, how add aria attribute in the code generated by the component (or directive ), look my pull request ?

In my advise, it's not possible ?
But I may be wrong, in any case I don't know how to do it ... 
@dmitry-zhemchugov  ???

File: src/typeahead/typeahead-container.component.ts [hunk 1]
```diff
@@ -92,6 +102,7 @@ export class TypeaheadContainerComponent implements OnDestroy {
     public element: ElementRef,
     private changeDetectorRef: ChangeDetectorRef
   ) {
+    this.element.nativeElement.id = this.popupId;
```
Comment (IraErshova) [inline src/typeahead/typeahead-container.component.ts]: Please use **renderer** to add new attribute

File: src/typeahead/typeahead.directive.ts [hunk 2]
```diff
@@ -140,6 +149,9 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
   /**  if false don't focus the input element the typeahead directive is associated with on selection */
     // @Input() protected typeaheadFocusOnSelect:boolean;
 
+  activeDescendant: string;
+  popupId: string;
```
Comment (IraErshova) [inline src/typeahead/typeahead.directive.ts]: Do you need this property? It looks like you don't use it in the directive, you use only `this._container.popupId `

File: src/typeahead/typeahead-container.component.ts (hunk 1 above)
Comment (samshay) [reply src/typeahead/typeahead-container.component.ts]: Ok, i make this

File: src/typeahead/typeahead.directive.ts (hunk 2 above)
Comment (samshay) [reply src/typeahead/typeahead.directive.ts]: it's right, i delete this property

**Review by IraErshova — DISMISSED** [review]

**Review by daniloff200 — APPROVED** [review]

**Comment by dmitry-zhemchugov** [comment]
Tested. Ready to megre

**Comment by daniloff200** [comment]
Finally, we did it!
Woohoo! :smile:  :wine_glass:  :champagne:

**Comment by samshay** [comment]
Thanks to @dmitry-zhemchugov @IraErshova @daniloff200 and @valorkin

**Comment by dharitdesai** [comment]
@dmitry-zhemchugov  @IraErshova  @daniloff200  @valorkin @samshay 
there is an issue in here, https://github.com/valor-software/ngx-bootstrap/blob/2c6a1f809e42127d6ba8abc97d9b1c6f15478fa1/src/typeahead/typeahead.directive.ts#L39

that `list` is not defined.
if you try to build your project by using this using AoT compiler, it throws error in build like this
`ERROR: src/location/location.html(16,9): : Directive TypeaheadDirective, Property 'list' does not exist on type 'TypeaheadDirective'.
`
I can throw a PR if you guys want to

**Pull request merged**